test/integration/src/main/java/io/pravega/test/integration/demo/interactive/BackgroundConsoleListener.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.integration.demo.interactive;
+
+import java.util.Scanner;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.val;
+
+/**
+ * Acquires console input on a background thread and updates its state ({@link #isTriggered()} when a desired token
+ * has been typed in.
+ */
+class BackgroundConsoleListener implements AutoCloseable {
```

Comment:
What about using JLine  ?

It  provides a few useful tools, like auto completion and history,

I have been playing with Pravega CLI during the last days and not having autocomplete and history is really awkward 

IMO for this very first release of the tool we can live without JLine.

But we should take a little bit of UX into consideration, especially for this kind of demo tools and are the first way of approaching the product by people who are deciding to adopt it.  


cc @derekm